cluster-autoscaler/core/utils/utils.go

```diff
@@ -91,15 +90,14 @@ func FilterOutNodesFromNotAutoscaledGroups(nodes []*apiv1.Node, cloudProvider cl
 }
 
 // DeepCopyNodeInfo clones the provided nodeInfo
-func DeepCopyNodeInfo(nodeInfo *schedulerframework.NodeInfo) *schedulerframework.NodeInfo {
-	newPods := make([]*apiv1.Pod, 0)
-	for _, podInfo := range nodeInfo.Pods {
-		newPods = append(newPods, podInfo.Pod.DeepCopy())
+func DeepCopyNodeInfo(nodeInfo *framework.NodeInfo) *framework.NodeInfo {
+	newPods := make([]*framework.PodInfo, 0)
+	for _, podInfo := range nodeInfo.Pods() {
+		newPods = append(newPods, &framework.PodInfo{Pod: podInfo.Pod.DeepCopy()})
 	}
 
 	// Build a new node info.
-	newNodeInfo := schedulerframework.NewNodeInfo(newPods...)
-	newNodeInfo.SetNode(nodeInfo.Node().DeepCopy())
+	newNodeInfo := framework.NewNodeInfo(nodeInfo.Node().DeepCopy(), nil, newPods...)
```

Comment:
Shouldn't this also copy DRA stuff?

Reply:
Yes, definitely. This will come in a later commit though, after these NodeInfo utils are refactored a bit: https://github.com/kubernetes/autoscaler/pull/7479. I'm trying to have these first PRs as just refactors so that we can get them in easier, and then have all of the DRA logic in the last PRs.